src/communication/handlers.ts

```diff
@@ -69,6 +70,34 @@ function buildNotification (agreement: Agreement, message: CommsMessage<CommsPay
   }
 }
 
+async function getAgreementWithRetry (
+  agreementRef: string,
+  retriesCount: number,
+  retryInterval: number
+): Promise<Agreement | undefined> {
+  function getAgreement (): Promise<Agreement> {
+    logger.info('Trying to get agreement ' + agreementRef)
+    return Agreement
+      .findOne({ where: { agreementReference: agreementRef } })
+      .then(agreement => {
+        if (!agreement) {
+          throw new Error('Agreement not found')
+        }
+        return agreement
+      })
+  }
+  return await asyncRetry<Agreement>(
+    getAgreement,
+    retriesCount,
+    retryInterval
+  ).catch(e => {
+    if (e.message.indexOf('Max retries reached for function') !== -1) {
```

Comment:
Uff this is not really good way to do this... Error's message should not be a way how you assert errors if you have any other way to do so and since this is our code, you can do it better ;-)

Messages are supposed to be for human not for code ;-) What if you change to wordings of the messages/fix a typo? Everything will brake...

Please create custom error like for example [here](https://github.com/rsksmart/rif-storage-pinner/blob/master/src/errors.ts#L38) and then use it like [here](https://github.com/rsksmart/rif-storage-pinner/blob/5590697fd7cd224b9a430bbcd2b3232066ebd389/src/gc.ts#L60)